download_plugins.py

```diff
@@ -1,40 +1,57 @@
-import requests, yaml, sys
+import requests
+import yaml
+import sys
+from distutils.version import LooseVersion
+
 
 # Get dependencies for a plugin, or get depedencies for a dependency.
 def get_dependencies(plugin):
-    print "Download dependencies for:", plugin
-    print "*****************************************"
-    print plugin, "has these dependencies:"
+    print("Finding dependencies for:", plugin)
+    print("*****************************************")
+    print(plugin, "has these dependencies:")
 
     try:
         dependencies = plugins_list['plugins'][plugin]['dependencies']
     except KeyError:
-        print "Unable to find dependencies for %s." % plugin
+        print("Unable to find dependencies for %s." % plugin)
+        global exit_code
         exit_code = 1
         return None
-    
+
     for dependency in dependencies:
-        print "Processing dependency: " + dependency['name']
-        
-        if (dependency['name'] not in installed_plugins) and (not dependency['optional']):
-            print "installing " + dependency['name']
+        print("Processing dependency: " + dependency['name'])
+
+        if dependency['optional']:
+                print(dependency['name'] + " is optional")
+
+        elif (dependency['name'] in stored_plugins):
+            if (LooseVersion(dependency['version']) <=
+                    LooseVersion(stored_plugins[dependency['name']])):
```

Comment:
Try to be consistent with [using quotes](https://www.python.org/dev/peps/pep-0008/?#string-quotes). You could choose either one (double quotes or single quotes) and stick to it.